src/DebugEngineHost.Stub/DebugEngineHost.ref.cs

```diff
@@ -197,56 +182,131 @@ public object GetCustomLauncher(string launcherTypeName)
+    /// <summary>
+    /// The channel used for logging messages.
+    /// Channels are used if there are multiple types of logs,
+    /// e.g. Engine logs and Natvis logs
+    /// </summary>
+    public interface ILogChannel
+    {
+        /// <summary>
+        /// Writes the given message with a newline to the log channel.
+        /// </summary>
+        /// <param name="level">The level of the log</param>
+        /// <param name="message">The message string to send.</param>
+        void WriteLine(LogLevel level, string message);
+
+        /// <summary>
+        /// Writes the given formatted message with the additional values with a newline to the log channel.
+        /// </summary>
+        /// <param name="level">The level of the log</param>
+        /// <param name="format">Format to use.</param>
+        /// <param name="values">Values to use within the provided format.</param>
+        void WriteLine(LogLevel level, string format, params object[] values);
+
+        /// <summary>
+        /// If the log is implemented as a file, this flushes the file.
+        /// </summary>
+        void Flush();
+
+        /// <summary>
+        /// If the log is implemented as a file, this closes the file.
+        /// </summary>
+        void Close();
+    }
+
+    /// <summary>
+    /// 
+    /// </summary>
+    public sealed class HostLogger
+    {
+        // EnableNatvisLogger is only used in OpenDebugAD7
+
+        /// <summary>
+        /// Enables engine logging if not already enabled.
+        /// </summary>
+        /// <param name="callback">The callback to use to send the engine log.</param>
+        /// <param name="level">The level of the log to filter the channel on.</param>
+        public static void EnableHostLogging(Action<string> callback, LogLevel level = LogLevel.Information)
         {
             throw new NotImplementedException();
         }
 
         /// <summary>
-        /// Writes a line to the log
+        /// Sets the log file to write to.
         /// </summary>
-        /// <param name="line">Line to write.</param>
-        public void WriteLine(string line)
+        /// <param name="logFile">The file to write engine logs to.</param>
+        public static void SetEngineLogFile(string logFile)
         {
             throw new NotImplementedException();
         }
 
+
         /// <summary>
-        /// If the log is implemented as a file, this flushes the file.
+        /// Gets the engine log channel created by 'EnableHostLogging'
         /// </summary>
-        public void Flush()
+        /// <returns></returns>
+        public static ILogChannel GetEngineLogChannel()
         {
             throw new NotImplementedException();
         }
 
         /// <summary>
-        /// If the log is implemented as a file, this closes the file.
+        /// Gets the Natvis log channel if its been created.
         /// </summary>
-        public void Close()
+        /// <returns></returns>
```

Comment:
><returns></returns> [](http://example.com/codeflow?start=12&length=19)

`<returns>A logger object if natvis logging is enabled, or null if it is not</returns>`